tests/installation/partitioning_togglehome.pm

```diff
@@ -29,7 +29,7 @@ sub run {
     }
     # For s390x there was no offering of separated home partition until SLE 15 See bsc#1072869
     if (!check_var('ARCH', 's390x') or is_storage_ng()) {
-        if (!check_screen 'disabledhome', 0) {
+        if (!check_screen 'disabledhome', 5) {
```

Comment:
Ii would really like to see the 0 kept to not waste waiting time in case the screen should not match. Did we maybe miss some waiting in before?

Reply:
I am not sure I understand the question :(  I have two examples of autoinst-log.txt with following extracts that show the timeout on check_screen("disabledhome")
 I kept the two files in https://michelmno.fedorapeople.org/PR5091_data/ if you want the full logs (hoping you have access)
```
=== failure1:
[2018-05-18T07:15:21.0826 CEST] [debug] /var/lib/openqa/share/tests/opensuse/tests/installation/partitioning_togglehome.pm:26 called testapi::assert_screen
[2018-05-18T07:15:21.0827 CEST] [debug] <<< testapi::assert_screen(mustmatch='partition-scheme', timeout=30)
...
[2018-05-18T07:15:22.0050 CEST] [debug] >>> testapi::_handle_found_needle: found partition-scheme-20180322, similarity 1.00 @ 563/357
[2018-05-18T07:15:22.0051 CEST] [debug] /var/lib/openqa/share/tests/opensuse/tests/installation/partitioning_togglehome.pm:27 called testapi::send_key
[2018-05-18T07:15:22.0051 CEST] [debug] <<< testapi::send_key(key='alt-n', do_wait=0)
[2018-05-18T07:15:22.0256 CEST] [debug] /var/lib/openqa/share/tests/opensuse/tests/installation/partitioning_togglehome.pm:32 called testapi::check_screen
[2018-05-18T07:15:22.0256 CEST] [debug] <<< testapi::check_screen(mustmatch='disabledhome', timeout=0)
...
[2018-05-18T07:15:23.0820 CEST] [debug] >>> testapi::_check_backend_response: match=disabledhome timed out after 0
[2018-05-18T07:15:23.0848 CEST] [debug] /var/lib/openqa/share/tests/opensuse/tests/installation/partitioning_togglehome.pm:34 called testapi::check_screen
[2018-05-18T07:15:23.0848 CEST] [debug] <<< testapi::check_screen(mustmatch='inst-partition-radio-buttons', timeout=0)
=== failure 2:
[2018-05-18T09:25:08.0904 CEST] [debug] /var/lib/openqa/share/tests/opensuse/tests/installation/partitioning_togglehome.pm:26 called testapi::assert_screen
[2018-05-18T09:25:08.0904 CEST] [debug] <<< testapi::assert_screen(mustmatch='partition-scheme', timeout=30)
...
[2018-05-18T09:25:09.0139 CEST] [debug] >>> testapi::_handle_found_needle: found partition-scheme-20180322, similarity 1.00 @ 563/357
[2018-05-18T09:25:09.0139 CEST] [debug] /var/lib/openqa/share/tests/opensuse/tests/installation/partitioning_togglehome.pm:27 called testapi::send_key
[2018-05-18T09:25:09.0139 CEST] [debug] <<< testapi::send_key(key='alt-n', do_wait=0)
[2018-05-18T09:25:09.0344 CEST] [debug] /var/lib/openqa/share/tests/opensuse/tests/installation/partitioning_togglehome.pm:32 called testapi::check_screen
[2018-05-18T09:25:09.0344 CEST] [debug] <<< testapi::check_screen(mustmatch='disabledhome', timeout=0)
...
[2018-05-18T09:25:10.0906 CEST] [debug] >>> testapi::_check_backend_response: match=disabledhome timed out after 0
[2018-05-18T09:25:10.0933 CEST] [debug] /var/lib/openqa/share/tests/opensuse/tests/installation/partitioning_togglehome.pm:34 called testapi::check_screen
[2018-05-18T09:25:10.0934 CEST] [debug] <<< testapi::check_screen(mustmatch='inst-partition-radio-buttons', timeout=0)
```

Reply:
There is a similar problem for aarch64:
https://openqa.opensuse.org/tests/679658#step/partitioning_togglehome/6

Reply:
> (hoping you have access)

Yes, I can access

> There is a similar problem for aarch64:
> https://openqa.opensuse.org/tests/679658#step/partitioning_togglehome/6

Not the same because this is the assert_screen further down which has the default timeout of 30s

My point is: Any check_screen with a timeout different than 0 seconds asks for race conditions. Because it might take 6s for a dialog to show up then `check_screen …, 5` is not enough but then people are afraid to raise the timeout to big values, like 30 seconds, because this would waste time in case the screen should actually not show up.

I see basically two options and this also shows as best practices we have found for other test modules:
1. A multi-tag assert_screen with use of `if (match_has_tag…`
2. A check_screen with 0 timeout

Option 1 can work in general when one action can trigger multiple outcomes. Option 2 works when on an *unchanged* screen one wants to check for more details.

What I can see in the code you want to change is that L23 is using `wait_screen_change` so it could be that the `wait_screen_change` triggers but the next screen is not fully loaded yet. So the change should rather be in this line to check for the right outcome after the key press of L23.

It seems the last code change that actually touched these lines was 142a5c372 by me.

I will see what I can do to improve the situation but I am happy if you want to take over.

Reply:
I also rechecked the above aarch64 failing test, its log confirms the failure is also caused by the timeout of check_screen(disabledhome) of line 32.
https://openqa.opensuse.org/tests/679658/file/autoinst-log.txt